.github/renovate.json5

```diff
@@ -41,10 +40,10 @@
         'react-router-dom',
       ],
       matchUpdateTypes: ['major'],
-      enabled: false,
+      "dependencyDashboardApproval": true
     },
     {
-      // Ignore major version bumps for these Ruby packages
+      // Require Dependency Dashboard Approval for major version bumps of these Ruby packages
       matchManagers: ['bundler'],
       matchPackageNames: [
         'rack', // Needs to be synced with Rails version
```

Comment:
For `rack`, we do not want to update it at all. This is an indirect dependency from `rails` (and maybe other), which depends on a specific major version of Rack.

I am wondering if we could remove it entirely, if Renovate is smart enough to detect that we can not bump it due to the major version constraint in `rails`?

Reply:
I created a preset that groups common Ruby on Rails dependencies into one PR, see:

- https://github.com/renovatebot/renovate/pull/17193

The current `group:rubyOnRails` preset put the Gems in the `matchPackageNames` array into a single Renovate PR:

```json
{
  "packageRules": [
    {
      "groupName": "Ruby on Rails packages",
      "matchDatasources": [
        "rubygems"
      ],
      "matchPackageNames": [
        "actioncable",
        "actionmailbox",
        "actionmailer",
        "actionpack",
        "actiontext",
        "actionview",
        "activejob",
        "activemodel",
        "activerecord",
        "activestorage",
        "activesupport",
        "railties",
        "rails"
      ]
    }
  ]
}
```

Do you think it's a good idea to add `rack` to the group? I don't use Ruby on Rails or `rack`, so I need your input here. 😉 

> I am wondering if we could remove it entirely, if Renovate is smart enough to detect that we can not bump it due to the major version constraint in `rails`?

I don't know. The quickest way to find out is to try it, and see what happens. 🙈

Reply:
I dont think so, `rack` is not a gem maintained by Rails (as all the ones you put in your group are), but major Rails versions depend on a specific major `rack` version (through `actionpack`).

The issue here is that we depend on `rack ~> 2.2.7`, as Rails does, but Renovate wants to push an update to `rack > 3`, which is maybe compatible with our Gemfile, but will not be compatible with ActionPack's version (`rack ~> 2.0, >= 2.2.4`)

So I would expect here that Renovate understand that it cant update our `rack` dependency, because one of our dependency's dependency will not support it. 

This is why I chose to have Renovate ignore `rack` major update completely, because we can only bump them to the next major when we upgrade Rails.